Improve ProximityCapturable

obrakmann opened the pull request:
The following changes are made:
- Replaced the custom actor search with ProximityTriggers.
- Added a `Sticky` flag, which causes the ownership to not revert back to
  the original owner when the captor leaves.
- Added an option to show a range circle for the actor, either constantly
  or on selection, drawn in the colour of the current owner.
  This will be especially useful for invisible ProximityCapturables.
- Added trait documentation.

obrakmann committed:
Improve ProximityCapturable

The following changes are made:

* Replaced the custom actor search with ProximityTriggers.
* Added a `Sticky` flag, which causes the ownership to not revert back to
  the original owner when the captor leaves.
* Added trait documentation.

abcdefg30 commented:
Looks good to me, and ingame worked fine, too. :+1: / :white_check_mark:

Mailaender commented on OpenRA.Mods.Common/Traits/ProximityCapturable.cs:
```diff
@@ -9,24 +9,44 @@
 
 namespace OpenRA.Mods.Common.Traits
 {
+	public enum RangeCircleVisibility { Never, Always, WhenSelected }
```
How about using `WithRangeCircle` instead or `RangeCircleRenderable.RenderDebugGeometry`?

obrakmann replied:
Fixing `WithRangeCircle` makes sense. I never quite liked its limitations. Updated.

@abcdefg30: I guess I'll have to remove your :+1: for that, sorry.

obrakmann committed:
Make WithRangeCircle a little more flexible

* Added ability to show the circle in the player's colour.
* Added ability to show the circle either always or only when selected.
* Added ability to configure the stances players need to see the circle.

Mailaender commented:
Can you rebase this please? https://github.com/OpenRA/OpenRA/issues/10548#issuecomment-172661671

obrakmann commented:
Rebased. Added a testcase as well, with which I found and fixed a few bugs.

The test case is Hope's Anchor. Harvester can be proximity-captured in this version (both `Sticky` and `MustBeClear` are on), and also the three strategic points have different characteristics:

* The left-most one: acts like a regular strategic point, but has a range circle when selected.
* The middle one: has the sticky behaviour, and `MustBeClear` is off. Also has the range circle on selection.
* The right-most one: Can be captured only once, and always shows the range circle.

Mailaender commented:
Works as promised. :white_check_mark:

abcdefg30 commented:
Looks good to me. :+1:  Remove the testcase, please.

obrakmann commented:
Test case removed. Let the record show that it can be found as commit b3794ff.

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/deedeb7c0f2d5fd49f76a596b281644b17046283...609de6edce5808ad4110c1be3862ddbe26892b47)
